BugSplatDotNetStandard/Api/VersionsClient.cs

```diff
@@ -0,0 +1,153 @@
+{
+    /// <summary>
+    /// Used to make requests to the BugSplat Versions API
+    /// </summary>
+    public class VersionsClient: IDisposable
+    {
+        private IBugSplatApiClient bugsplatApiClient;
+        private S3Client s3Client;
+
+        internal VersionsClient(
+            IBugSplatApiClient bugsplatApiClient,
+            IS3ClientFactory s3ClientFactory
+        )
+        {
+            ThrowIfArgumentIsNull(bugsplatApiClient, "bugsplatApiClient");
+            ThrowIfArgumentIsNull(s3ClientFactory, "s3ClientFactory");
+
+            this.bugsplatApiClient = bugsplatApiClient;
+            this.s3Client = s3ClientFactory.Create();
+        }
+
+        /// <summary>
+        /// Create an unauthenticated VersionsClient
+        /// </summary>
+        /// <param name="clientId">BugSplat OAuth2 ClientId</param>
+        /// <param name="clientSecret">BugSplat OAuth2 ClientId</param>
+        public static VersionsClient Create(IBugSplatApiClient bugsplatApiClient)
+        {
+            return new VersionsClient(bugsplatApiClient, S3ClientFactory.Default);
+        }
+
+        /// <summary>
+        /// Get a list of applications and versions with crashes and/or symbols for a BugSplat database
+        /// </summary>
+        /// <param name="database">BugSplat database for which version information will be retrieved</param>
+        public async Task<HttpResponseMessage> GetVersions(string database)
+        {
+            ThrowIfArgumentIsNullOrEmpty(database, "database");
+
+            ThrowIfNotAuthenticated(bugsplatApiClient);
+
+            var response = await this.bugsplatApiClient.GetAsync($"/api/versions?database={database}");
+
+            ThrowIfHttpRequestFailed(response);
+
+            return response;
+        }
+
+        /// <summary>
+        /// Upload a symbol file to BugSplat and authenticate if not already authenticated
+        /// </summary>
+        /// <param name="database">BugSplat database to upload symbols too</param>
+        /// <param name="application">BugSplat application value to associate with the symbol file</param>
+        /// <param name="version">BugSplat version value to associate with the symbol file</param>
+        /// <param name="symbolFileInfo">The symbol file that will be uploaded to BugSplat</param>
+        public async Task<HttpResponseMessage> UploadSymbolFile(
+            string database,
+            string application,
+            string version,
+            FileInfo symbolFileInfo
+        )
+        {
+            ThrowIfArgumentIsNullOrEmpty(database, "database");
+            ThrowIfArgumentIsNullOrEmpty(application, "application");
+            ThrowIfArgumentIsNullOrEmpty(version, "version");
+            ThrowIfArgumentIsNull(symbolFileInfo, "symbolFileInfo");
+
+            ThrowIfNotAuthenticated(bugsplatApiClient);
+
+            var random = Path.GetRandomFileName();
+            var zipFileName = $"bugsplat-{symbolFileInfo.Name}-{random}.zip";
```

Comment:
Let's add an issue for getting the GUID from the PDB! Let's turn https://github.com/BugSplat-Git/src-backend/blob/799208e53c3d8a13bf35c3ba1c405ecdc81a4c6a/BugSplat/PdbLibrary/PdbLibrary/PdbLibrary.cs#L295 into a common package and consume it here!

Reply:
#53